Libs/MRML/Widgets/qMRMLNodeComboBox.cxx

```diff
@@ -695,16 +708,44 @@ void qMRMLNodeComboBox::renameCurrentNode()
     return;
   }
 
+  // Use the dialog method to get the new name from the user
   bool ok = false;
-  QString newName = QInputDialog::getText(
-    this, "Rename " + this->nodeTypeLabel(node->GetClassName()), "New name:",
-    QLineEdit::Normal, node->GetName(), &ok);
-  if (!ok)
+  QString currentName = node->GetName();
+  QString newName = getNameFromDialog(renameTitlePrefix + this->nodeTypeLabel(node->GetClassName()), currentName, &ok);
+
+  // Check if the user confirmed the action
+  if (ok && !newName.isEmpty())
   {
-    return;
+    node->SetName(newName.toUtf8());
+    emit currentNodeRenamed(newName);
   }
-  node->SetName(newName.toUtf8());
-  emit currentNodeRenamed(newName);
+}
+// --------------------------------------------------------------------------
+void qMRMLNodeComboBox::createNodeAs(const QString& nodeTypeName)
+{
+    QString label = this->nodeTypeLabel(nodeTypeName);
+    bool okPressed = false;
+    QString dialogTitle = tr("Create New %1").arg(label);
+    QString defaultName = "";
+    QString nodeName = this->getNameFromDialog(dialogTitle, defaultName, &okPressed);
+
+    if (okPressed && !nodeName.isEmpty())
+    {
+        vtkMRMLNode* newNode = this->addNode(nodeTypeName);
+        if (newNode != nullptr)
+        {
+            newNode->SetName(nodeName.toUtf8());
+            this->setCurrentNode(newNode);
+            emit this->nodeAddedByUser(newNode);
+        }
+    }
```

Comment:
It seems like this is the key change to fix the reported issue.  Maybe it should be a dedicated commit and the i18n changes would be independent.

Or if the i18n changes at least make it better than it was before we could integrate these and then make further improvements in another PR.

Reply:
If you think this is essencial, can you please explain the point?
Because I don't see any specific reason of doing this, this will take me extra effort to separate the commits. I also followed the contribution guidelines and did not notice such a requirement to separate different changes.